doc/user/using-java9.md

```diff
@@ -0,0 +1,34 @@
+# Using TruffleRuby with JDK 9
+
+TruffleRuby is designed to be run with a JVM that has the Graal compiler. The
+easiest way to do this is to use the GraalVM, which includes a JDK, the Graal
+compiler, and TruffleRuby, all in one package.
+
+It is also possible to run on Java 9, but Graal needs to be built with Java 9
+and there is no binary distribution of GraalVM based on Java 9 so you will
+need to [build Graal yourself](../contributor/building-graal.md), using Java
+9 as the JDK.
+
+```
+$ GRAAL_HOME=...path to graal repository built with JDK 9...
+$ JAVACMD=...java 9...
+$ bin/truffleruby \
+    -J-XX:+UnlockExperimentalVMOptions \
+    -J-XX:+EnableJVMCI \
+    -J--module-path=$GRAAL_HOME/sdk/mxbuild/modules/org.graalvm.graal_sdk.jar:$GRAAL_HOME/truffle/mxbuild/modules/com.oracle.truffle.truffle_api.jar:/$GRAAL_HOME/compiler/mxbuild/modules/jdk.internal.vm.compiler.jar \
+    -e 'p Truffle.graal?'
+```
```

Comment:
I think we should have a --java9 flag in the launcher or maybe even detect it automatically based in JAVACMD.